source/connection_manager.c

```diff
@@ -335,25 +366,40 @@ static void s_aws_http_connection_manager_complete_acquisitions(
         struct aws_http_connection_acquisition *pending_acquisition =
             AWS_CONTAINER_OF(node, struct aws_http_connection_acquisition, node);
 
-        pending_acquisition->callback(
-            pending_acquisition->connection, pending_acquisition->error_code, pending_acquisition->user_data);
-
         if (pending_acquisition->error_code != 0) {
             AWS_LOGF_WARN(
                 AWS_LS_HTTP_CONNECTION_MANAGER,
                 "id=%p: Failed to completed connection acquisition with error_code %d(%s)",
                 (void *)pending_acquisition->manager,
                 pending_acquisition->error_code,
                 aws_error_str(pending_acquisition->error_code));
+            pending_acquisition->callback(
```

Comment:
Can you guarantee here that `aws_channel_thread_is_callers_thread`?

Reply:
we don't actually care in this case.